api/api.go

```diff
@@ -137,6 +137,7 @@ func HandleAddJob(cache job.JobCache, defaultOwner string) func(http.ResponseWri
 
 		err = newJob.Init(cache)
 		if err != nil {
+			// TODO(itstehkman): should we run on_failure job here?
```

Comment:
I'd say not. If you can't initialize, that usually means something fundamentally wrong.